open-apparel-registry/open-apparel-registry
# Update user model

## Overview

This PR updates the User and Organization models to have them have the fields presented in #170 and updates the user registration handler endpoint to match.

In addition, the changes:

- adjust the registration endpoint to make the free text description for "Other Contributor Type" required when "Other" had been selected
- update the makefixtures command and standard fixtures to move name, description, org_type, and other_org_type to the Organization data
- update the tests to account for `name` having been moved off the User model.

Connects #170 
Connects #135 

## Testing Instructions

- get this branch, then `./scripts/manage reset_schema`, `./scripts/manage migrate` and `./scripts/manage loadfixtures`
- through the UI, register for a new account and select "Other" as the contributor type, leaving the other contributor type free text field blank but completing all other required fields. Verify that you see an error message indicating that the free text description is required
- add a description, then save the account and verify that it saves correctly
- register another new account but this time select "Contributor Type" of "Auditor" and verify that it saves as expected
- verify that the new model matches what's described in #170
